src/main/java/org/broadinstitute/hellbender/utils/collections/IntervalsSkipListOneContig.java

```diff
@@ -32,15 +33,16 @@
     // reach: bucket# -> how far that bucket reaches.
     // e.g. bucket 0 contains the first 2**shift locatables. reach[0] is the max over their .getEnd()
     //      reach[x] is the max over the .getEnd for that bucket and all the ones before it.
-    private int[] reach;
+    private final int[] reach;
+    private final int reachLength;
 
     /**
      * Creates an IntervalsSkipList that holds a copy of the given intervals, sorted
      * and indexed.
      *
      * @param loc Locatables, not necessarily sorted. Will be iterated over exactly once.
      */
-    public IntervalsSkipListOneContig(Iterable<T> loc) {
+    public IntervalsSkipListOneContig(final Iterable<T> loc) {
```

Comment:
It seems like this class should be package scoped, it's really an implementation detail of the `IntervalsSkipList`